src/katcbf_vlbi_resample/cupy_bridge.py

```diff
@@ -16,26 +16,42 @@
 
 """Convert streams to and from cupy."""
 
+import asyncio
 from collections import deque
-from collections.abc import Iterator
+from collections.abc import AsyncIterator
 
 import cupy as cp
 import cupyx
+import packaging.version
 import xarray as xr
 
 from .stream import ChunkwiseStream, Stream
-from .utils import as_cupy
+from .utils import as_cupy, stream_future
+
+# We don't list cupy in project requirements because there are multiple
+# packages that could provide it (e.g. cupy-cudaNNx for binary wheels),
+# so we check the dependencies at import time.
+_cupy_version = packaging.version.Version(cp.__version__)
+if _cupy_version < packaging.version.Version("13.3"):
+    raise ImportError("cupy >= 13.4 is required", name="cupy")
```

Comment:
From [this comment](https://github.com/ska-sa/katcbf-vlbi-resample/pull/23#discussion_r2378825498)
- Genuinely asking, why did you leave the packaging.version.Version arg as "13.3"?
- Can/should it be updated to "13.4"?

Reply:
Whoops, that's a mistake. I'll fix.